src/Context.cpp

```diff
@@ -446,8 +446,9 @@ int Context::initialize (int argc, const char** argv)
     ////////////////////////////////////////////////////////////////////////////
     //
     // [1] Load the correct config file.
-    //     - Default to ~/.taskrc (ctor).
+    //     - Default to ~/.config/taskrc (ctor).
     //     - Allow $TASKRC override.
+    //     - Allow $XDG_CONFIG_HOME/taskwarrior/taskrc override.
```

Comment:
Whoops I got the comments wrong here. Will fix, sorry for confusion.

I agree with the strategy you just described, and initially I tried my hand at implementing just that. But the entire default config is hardcoded here https://github.com/GothenburgBitFactory/taskwarrior/blob/2.6.0/src/Context.cpp#L60, and the code for getOverride https://github.com/GothenburgBitFactory/taskwarrior/blob/2.6.0/src/CLI2.cpp#L239 doesn't look like it lends itself very well to this strategy - let me think it over and see if there's a clean way to do this but I'm not sure there is without refactoring.

If you have any suggestions, I'm happy to hear them ofc. 

Also, of note is that currently the command-line argument overrides the TASKRC env var and not the opposite. The docs got it wrong too, btw https://github.com/GothenburgBitFactory/taskwarrior/pull/2316/files#diff-cf9f17ad6afd5b73870e3912a25c8999L157 I fixed them here. It could be its own commit or PR, let me know your thoughts if you have any on the matter.

Reply:
Update: this issue should now be resolved with the latest changes